Add localisation support for Skin Editor components

ItsShamed opened the pull request:
Not sure about the class structure and naming, but since some strings make more sense in such context, I decided to go with that.

ItsShamed committed:
feat: add localisation for Skin editor components

ItsShamed committed:
refactor: fix typos

ItsShamed committed:
Merge branch 'ppy:master' into locale/skin-editor-components

ItsShamed committed:
refactor: change namespacing to allow osu-localisation-analyzer to process all strings

ItsShamed committed:
quality: remove new line

bdach commented:
This is going to need a merge of, or a rebase on top of, https://github.com/ppy/osu/pull/22490 (merge conflicts due to namespace move) and https://github.com/ppy/osu/pull/22494 (new localisable strings).

In general this is maybe one of the worst possible times to be PRing this given that it looks that there is going to be considerable movement in the skin editor in the coming days/weeks.

ItsShamed commented:
Yeah, my jaw dropped when I saw peppy touching the `SkinEditor` during his livestream lmao


Guess I'll have to wait on how things goes, or maybe I'll open a PR once things are done with a cleaner commit history. Things don't really touch `SkinEditor` anyway, it's not that bad.

ItsShamed commented:
Yeah, my jaw dropped when I saw peppy touching the `SkinEditor` during his livestream lmao


Guess I'll have to wait on how things goes, or maybe I'll open a PR once things are done with a cleaner commit history. Things don't really touch `SkinEditor` anyway, it's not that bad.

ItsShamed commented:
Yeah, my jaw dropped when I saw peppy touching the `SkinEditor` during his livestream lmao


Guess I'll have to wait on how things goes, or maybe I'll open a PR once things are done with a cleaner commit history. Things don't really touch `SkinEditor` anyway, it's not that bad.

ItsShamed commented:
Yeah, my jaw dropped when I saw peppy touching the `SkinEditor` during his livestream lmao


Guess I'll have to wait on how things goes, or maybe I'll open a PR once things are done with a cleaner commit history. Things don't really touch `SkinEditor` anyway, it's not that bad.

peppy commented:
Might as well get this in while we can. My only apprehension is that I'm not sure how much of the structure of the skin editor I'm going to be changing (and how many of the localised strings may get changed or no longer be required).

peppy committed:
Merge branch 'master' into locale/skin-editor-components

peppy committed:
Fix incorrectly applied enum localisation

peppy committed:
Move skin component localisations to namespaces

peppy commented on osu.Game/Overlays/SkinEditor/SkinEditor.cs:
```diff
@@ -256,13 +256,13 @@ private void skinChanged()
             headerText.AddParagraph(SkinEditorStrings.SkinEditor, cp => cp.Font = OsuFont.Default.With(size: 16));
             headerText.NewParagraph();
-            headerText.AddText("Currently editing ", cp =>
+            headerText.AddText(SkinEditorStrings.CurrentlyEditing, cp =>
```
This may not work great with all languages, but I'm not sure we can do better currently if we want the formatting to remain as it is here.

peppy committed:
Combine localisations for abstract skinnable components into a single file

Generally we don't want localisation files with only one to two
translations. It makes it harder for translators to handle in crowdin (a
lot of file changes for small results).

So for cases like this I believe we should be grouping translations
where it makes sense.

I've left individual components in their own files as I can see
potential for more settings to be added in the future. Plus it gives a
bit of extra context.

Loreos7 commented:
You've missed the last piece of this pack: Horizontal/Vertical counter direction strings.
![image](https://github.com/ppy/osu/assets/86934170/e137b6e6-3f8e-437e-84bd-0be8901b5d13)

ItsShamed commented:
IIRC `Direction` is an enum from osu-framework, this needs to be addressed there and was out of scope for this PR.
